infra/cifuzz/clusterfuzz_deployment.py

```diff
@@ -0,0 +1,216 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module for interacting with the "ClusterFuzz deployment."""
+import logging
+import os
+import sys
+import tempfile
+import time
+import urllib.error
+import urllib.request
+import zipfile
+
+# pylint: disable=wrong-import-position,import-error
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+import utils
+
+
+class BaseClusterFuzzDeployment:
+  """Base class for ClusterFuzz deployments."""
+
+  def __init__(self, config):
+    self.config = config
+
+  def download_latest_build(self, out_dir):
+    """Downloads the latest build from ClusterFuzz.
+
+    Returns:
+      A path to where the OSS-Fuzz build was stored, or None if it wasn't.
+    """
+    raise NotImplementedError('Child class must implement method.')
+
+  def download_corpus(self, target_name, out_dir):
+    """Downloads the corpus for |target_name| from ClusterFuzz to |out_dir|.
+
+    Returns:
+      A path to where the OSS-Fuzz build was stored, or None if it wasn't.
+    """
+    raise NotImplementedError('Child class must implement method.')
+
+
+class ClusterFuzzLite(BaseClusterFuzzDeployment):
```

Comment:
Is this referring to the CIFuzz batch mode you added?

Reply:
Not quite, it's referring to a mode of running CIFuzz where CIFuzz is doing things like batch fuzzing, corpus pruning and coverage reports. So instead of having the OSS-Fuzz instance provide things like corpus (backups) and coverage reports it will provided by CIFuzz (aka ClusterFuzzLite).